service/metadata/service/permission.go

```diff
@@ -0,0 +1,260 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/math"
+	chaintypes "github.com/bnb-chain/greenfield/types"
+	"github.com/bnb-chain/greenfield/types/resource"
+	gnfdresource "github.com/bnb-chain/greenfield/types/resource"
+	permtypes "github.com/bnb-chain/greenfield/x/permission/types"
+	"github.com/bnb-chain/greenfield/x/storage/keeper"
+	storagetypes "github.com/bnb-chain/greenfield/x/storage/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/forbole/juno/v4/common"
+
+	"github.com/bnb-chain/greenfield-storage-provider/model/errors"
+	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
+	"github.com/bnb-chain/greenfield-storage-provider/store/bsdb"
+)
+
+// VerifyPermission Verify the input account’s permission to input items
+func (metadata *Metadata) VerifyPermission(ctx context.Context, req *storagetypes.QueryVerifyPermissionRequest) (resp *storagetypes.QueryVerifyPermissionResponse, err error) {
+	var (
+		operator   sdk.AccAddress
+		bucketInfo *bsdb.Bucket
+		objectInfo *bsdb.Object
+		effect     permtypes.Effect
+	)
+
+	ctx = log.Context(ctx, req)
+
+	if req == nil {
+		log.CtxErrorw(ctx, "invalid request", "error", err)
+		return nil, errors.ErrInvalidParams
+	}
+
+	operator, err = sdk.AccAddressFromHexUnsafe(req.Operator)
+	if err != nil {
+		log.CtxErrorw(ctx, "failed to creates an AccAddress from a HEX-encoded string", "error", err)
+		return nil, err
+	}
+
+	if req.BucketName == "" {
+		log.Errorw("failed to check bucket name", "bucket_name", req.BucketName, "error", err)
+		return nil, errors.ErrInvalidBucketName
+	}
+
+	bucketInfo, err = metadata.bsDB.GetBucketByName(req.BucketName, true)
+	if err != nil || bucketInfo == nil {
+		log.CtxErrorw(ctx, "failed to get bucket info", "error", err)
+		return nil, errors.ErrNoSuchBucket
+	}
+
+	if req.ObjectName == "" {
```

Comment:
`s3util.CheckValidObjectName`?

Reply:
This if statement is used to determine whether the user has passed in an object name rather than to check whether the object name is valid(compare to the above if statement)